lua/sim/weapons/DefaultBeamWeapon.lua

```diff
@@ -31,18 +31,23 @@ DefaultBeamWeapon = ClassWeapon(DefaultProjectileWeapon) {
             return
         end
 
+        -- Store common data for creating beams
+        local beamData = {
```

Comment:
I would call it a spec ("specification") since it's a special table used by the engine to initialize the entity.
```suggestion
        ---@type CollisionBeamSpec
        local beamSpec = {
```
Also it seems that the comments for the annotations of `BeamBone` and `OtherBone` are swapped.
https://github.com/FAForever/fa/blob/345f5ca5d3f9290a8128619e30dd5f6aac9535a6/engine/Sim/CollisionBeamEntity.lua#L5-L6

Reply:
Sounds good! I'll make that change before a merge.